#! /bin/sh -e
## 14_fix_geoip_warning.dpatch by  <weasel@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Change geoipdb open failed message

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' tor~/src/or/geoip.c tor/src/or/geoip.c
--- tor~/src/or/geoip.c	2012-09-13 23:13:52.000000000 +0200
+++ tor/src/or/geoip.c	2012-09-19 10:32:40.869252859 +0200
@@ -207,7 +207,8 @@
   crypto_digest_t *geoip_digest_env = NULL;
   clear_geoip_db();
   if (!(f = tor_fopen_cloexec(filename, "r"))) {
-    log_fn(severity, LD_GENERAL, "Failed to open GEOIP file %s.  %s",
+    log_fn(severity, LD_GENERAL, "Failed to open GEOIP file %s.  %s"
+           " Do you have the tor-geoipdb package installed?",
            filename, msg);
     return -1;
   }
